src/traces/scattergeo/plot.js

```diff
@@ -125,83 +125,84 @@ plotScatterGeo.plot = function(geo, scattergeoData) {
 
     gScatterGeoTraces.exit().remove();
 
-    // TODO add hover - how?
-    gScatterGeoTraces
-        .each(function(trace) {
-            if(!subTypes.hasLines(trace)) return;
+    // TODO find a way to order the inner nodes on update
+    gScatterGeoTraces.selectAll('*').remove();
```

Comment:
I have a few ideas that would allow us to remove this dirty purge call.

See:
- http://bl.ocks.org/etpinard/c163b8bee7d3bcb6a31f
- http://bl.ocks.org/etpinard/fb082e67cd821b4362ef

for some example. But I want to benchmark these patterns before implementing them in the code. So I opted for the easy route in this PR.